### ISSUE TITLE
False positive with count associtiaons

### ISSUE THREAD

**Issue by CarlosUvaSilva** [flyerhzm/bullet#642]
I have this inside a controller

```
users = User.includes(
  :posts,
)
```

and this inside the User model
```
def number_of_posts
  if posts.loaded?
    posts.size
  else
    posts.count
  end
end
```

If I include posts I get the expected:
`SELECT "posts".* FROM "posts" WHERE "posts"."user_id" IN ($1, $2, $3, $4, $5)`

and if I don't include I get one of these for each user:
`SELECT COUNT(*) FROM "posts" WHERE "posts"."user_id" = $1`

Problem is Bullet is telling me to NOT include the posts
```
AVOID eager loading detected
  User => [:posts]
  Remove from your query: .includes([:posts])
```

Thank you!

**Comment by shivabhusal** [comment]
I am also seeing massive false positives during upgrading Rails 6 to 7(bullet 7).

**Comment by full-of-foo** [comment]
Same here

**Comment by flyerhzm** [comment]
I think it supposed to be fixed in bullet 7.1.3

**Comment by full-of-foo** [comment]
I can confirm I'm still getting many false n+1's with Bullet 7.1.3 with Rails 7.0 and Rails 7.1 which were not present before upgrading to Rails 7.0 and 7.1. Hmmmm, @flyerhzm, any ideas 🤔 ?
